ffs/filcold/filcold.go

```diff
@@ -244,6 +279,28 @@ func (fc *FilCold) renewDeal(ctx context.Context, c cid.Cid, size uint64, p ffs.
 // makeDeals starts deals with the specified miners. It returns a slice with all the ProposalCids
 // that were started successfully, and a slice of DealError with deals that failed to be started.
 func (fc *FilCold) makeDeals(ctx context.Context, c cid.Cid, size uint64, cfgs []deals.StorageDealConfig, fcfg ffs.FilConfig) ([]cid.Cid, []ffs.DealError, error) {
+	fc.l.Log(ctx, "Entering deal execution queue...")
+	// In the next Lotus release (v1.1.3), we'll be able to remove this limiting
+	// since we can avoid creating deals recompute the deal size and CommP. Since
+	// that isn't ready yet, starting a deal is very resource intensive so put a cap for now.
+	select {
+	case fc.semaphDealPrep <- struct{}{}:
+	case <-ctx.Done():
+		return nil, nil, fmt.Errorf("canceled by context")
+	}
+	defer func() { <-fc.semaphDealPrep }()
+	for {
+		if fc.lsm.SyncHeightDiff() < unsyncedThreshold {
+			break
+		}
+		log.Warnf("lotus backpressure from unsynced node")
+		select {
+		case <-ctx.Done():
+			return nil, nil, fmt.Errorf("canceled by context")
+		case <-time.After(time.Minute):
+		}
+	}
```

Comment:
Similar limiting done in `calculatePieceSize`. I decided not to extract the limiting logic since this particular piece of code will go away in Lotus v1.1.3 when we have the new mentioned API.

When that lands, we don't have to rate limit things here... since we'll use what was calculated in `calculatePieceSize` when making deals, so Lotus doesn't have to recalculate heavy things, thus no need to rate-limit here anymore.